robotframework-ls/src/robotframework_ls/impl/flow_explorer_model_builder.py

```diff
@@ -4,6 +4,22 @@
 from robocorp_ls_core.basic import isinstance_name
 
 
+class _RecursionStack:
+    stack: List[str] = []
```

Comment:
You want to set the `stack` under the constructor (i.e.: `in def __init__(self)`) -- as it is any instance will use the same stack because it's declared in the class scope.

Also, it should probably be at least protected (i.e.: `_stack`).